### ISSUE TITLE
[zlib] build failure

### ISSUE THREAD

**Issue by samsaga2** [microsoft/vcpkg#13161]
I am trying to install zlib port with the last vcpkg version and I have errors. With an older vcpkg version I have no problems but I really need to update some ports.

Everytime I update vcpkg one thing or another is broken. I am really thinking to migrate to another system.

```
.\vcpkg.exe install zlib:x64-windows --debug
```

```
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Using vcpkg-root: D:\dev\vcpkg
[DEBUG] Using installed-root: D:\dev\vcpkg\installed
[DEBUG] Using buildtrees-root: D:\dev\vcpkg\buildtrees
[DEBUG] Using downloads-root: D:\dev\vcpkg\downloads
[DEBUG] Using packages-root: D:\dev\vcpkg\packages
[DEBUG] Using scripts-root: D:\dev\vcpkg\scripts
[DEBUG] Default binary cache path is: C:\Users\VictorM\AppData\Local\vcpkg\archives
Computing installation plan...
[DEBUG] Found path: C:\Program Files\CMake\bin\cmake.exe
[DEBUG] CreateProcessW("C:\Program Files\CMake\bin\cmake.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    22230 us
[DEBUG] CreateProcessW("C:\Program Files\CMake\bin\cmake.exe" -DVCPKG_ROOT_DIR=D:/dev/vcpkg -DPACKAGES_DIR=D:/dev/vcpkg/packages -DBUILDTREES_DIR=D:/dev/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=D:/dev/vcpkg/installed -DDOWNLOADS=D:/dev/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "D:\dev\vcpkg\buildtrees\0.vcpkg_tags.cmake")
[DEBUG] cmd_execute_and_stream_data() returned 0 after    15778 us
The following packages will be built and installed:
    zlib[core]:x64-windows
[DEBUG] CreateProcessW("C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -legacy -products * -format xml)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    18236 us
Detecting compiler hash for triplet x64-windows...
[DEBUG] Found path: C:\Program Files\SmartGit\git\bin\git.exe
[DEBUG] CreateProcessW("D:\dev\vcpkg\downloads\tools\git-2.26.2-1-windows\mingw32\bin\git.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    13080 us
[DEBUG] CreateProcessW(cmd /c ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" -vcvars_ver=14.2 amd64   2>&1 <NUL" & echo cdARN4xjKueKScMy9C6H& set)
[DEBUG] cmd_execute_and_stream_data() returned 0 after   902873 us
[DEBUG] CreateProcessW("C:\Program Files\CMake\bin\cmake.exe" -DCURRENT_PORT_DIR=D:/dev/vcpkg/scripts/detect_compiler -DCURRENT_BUILDTREES_DIR=D:/dev/vcpkg/buildtrees/detect_compiler -DCURRENT_PACKAGES_DIR=D:/dev/vcpkg/packages/detect_compiler_x64-windows -DCMD=BUILD -DTARGET_TRIPLET=x64-windows "-DTARGET_TRIPLET_FILE=D:\dev\vcpkg\triplets\x64-windows.cmake" -DVCPKG_PLATFORM_TOOLSET=v142 -DDOWNLOADS=D:/dev/vcpkg/downloads -DVCPKG_CONCURRENCY=9 -DGIT=D:/dev/vcpkg/downloads/tools/git-2.26.2-1-windows/mingw32/bin/git.exe -DVCPKG_ROOT_DIR=D:/dev/vcpkg -DPACKAGES_DIR=D:/dev/vcpkg/packages -DBUILDTREES_DIR=D:/dev/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=D:/dev/vcpkg/installed -DDOWNLOADS=D:/dev/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "D:\dev\vcpkg\scripts\ports.cmake")
[DEBUG] -- Configuring x64-windows
[DEBUG] [1/1] cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/DEV/vcpkg/scripts/detect_compiler" "-DCMAKE_MAKE_PROGRAM=C:/Windows/System32/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/DEV/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/DEV/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/dev/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/dev/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/dev/vcpkg/packages/detect_compiler_x64-windows""
[DEBUG] #COMPILER_HASH#12e212886f6f3928abf1d2d3d5106df42e436722
[DEBUG] -- Configuring done
[DEBUG] -- Generating done
[DEBUG] CMake Warning:
[DEBUG]   Manually-specified variables were not used by the project:
[DEBUG] 
[DEBUG]     BUILD_SHARED_LIBS
[DEBUG]     CMAKE_INSTALL_BINDIR
[DEBUG]     CMAKE_INSTALL_LIBDIR
[DEBUG]     VCPKG_TARGET_ARCHITECTURE
[DEBUG] 
[DEBUG] 
[DEBUG] -- Build files have been written to: D:/DEV/vcpkg/buildtrees/detect_compiler/x64-windows-rel
[DEBUG] 
[DEBUG] cmd_execute_and_stream_data() returned 0 after   924966 us
[DEBUG] 
[DEBUG] Detecting compiler hash for triplet x64-windows: 12e212886f6f3928abf1d2d3d5106df42e436722
[DEBUG] <abientries for zlib:x64-windows>
[DEBUG]   CONTROL|94be6280da4863610bf069e4edf0af3e7db31c99
[DEBUG]   LICENSE|e23c6e5f40647ee2f817439d3858531fb0753c9d
[DEBUG]   cmake|3.18.2
[DEBUG]   cmake_dont_build_more_than_needed.patch|380b131349bf645f05637612e4a9a04ed8ba6349
[DEBUG]   features|core
[DEBUG]   portfile.cmake|2f85d9d5861bef273b2b68149e895109b5140334
[DEBUG]   post_build_checks|2
[DEBUG]   powershell|ab1effc926b000a6adc12198a1886514ec203621a53b0cd7ec1cd9a8225dccda7e857feaabcfba4004bea73129b986abaad777c4573f44e0af70411226ce08b0
[DEBUG]   triplet|3eab5c64d9cf628fc0aec277fde66dbdc69b8516-83a62cd5c237de1a32ff2f55ead5e6b66c461200-12e212886f6f3928abf1d2d3d5106df42e436722
[DEBUG]   usage|b446f7466dead962527355c13bbb617213098938
[DEBUG]   vcpkg_configure_cmake|1264ee650203750d3c91272e9da8f8d9de0b948d
[DEBUG]   vcpkg_copy_pdbs|4bbc15802daf8f9f8a92794c10b0a89cfca1e742
[DEBUG]   vcpkg_download_distfile|464118e33fefa6fe513912a900cdf70ebe6b62d2
[DEBUG]   vcpkg_extract_source_archive|40d93d67b03dd376ba314638bcfbd0f9e6e5a930
[DEBUG]   vcpkg_extract_source_archive_ex|82c579269d8f58679b16440d3d7d98be4e024853
[DEBUG]   vcpkg_fixup_pkgconfig|7999b090e55aef0866dda59c6b167c52b55467f5
[DEBUG]   vcpkg_install_cmake|75d1079d6f563d87d08c8c5f63b359377aa19a6e
[DEBUG]   vcpkg_replace_string|99deefc316db487a541ba0b25d6249a0b86e2186
[DEBUG]   vcpkg_test_cmake|be514c057fc9fda9c64dda13ab2c81ccd0be87fe
[DEBUG] </abientries>
Starting package 1/1: zlib:x64-windows
Building package zlib[core]:x64-windows...
Could not locate cached archive: C:\Users\VictorM\AppData\Local\vcpkg\archives\9e\9e85866f42beda8bb97a47c5fb6ac37f031faeb9.zip
[DEBUG] CreateProcessW("C:\Program Files\CMake\bin\cmake.exe" -DPORT=zlib -DCURRENT_PORT_DIR=D:/dev/vcpkg/ports/zlib -DVCPKG_USE_HEAD_VERSION=0 -D_VCPKG_NO_DOWNLOADS=0 -D_VCPKG_DOWNLOAD_TOOL=BUILT_IN -D_VCPKG_EDITABLE=0 -DFEATURES=core -DALL_FEATURES= -DCMD=BUILD -DTARGET_TRIPLET=x64-windows "-DTARGET_TRIPLET_FILE=D:\dev\vcpkg\triplets\x64-windows.cmake" -DVCPKG_PLATFORM_TOOLSET=v142 -DDOWNLOADS=D:/dev/vcpkg/downloads -DVCPKG_CONCURRENCY=9 -DGIT=D:/dev/vcpkg/downloads/tools/git-2.26.2-1-windows/mingw32/bin/git.exe -DVCPKG_ROOT_DIR=D:/dev/vcpkg -DPACKAGES_DIR=D:/dev/vcpkg/packages -DBUILDTREES_DIR=D:/dev/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=D:/dev/vcpkg/installed -DDOWNLOADS=D:/dev/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "D:\dev\vcpkg\scripts\ports.cmake")
-- Using cached D:/dev/vcpkg/downloads/zlib1211.tar.gz
-- Cleaning sources at D:/dev/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/dev/vcpkg/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Using source at D:/dev/vcpkg/buildtrees/zlib/src/1.2.11-deec42f53b.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Fixing pkgconfig
-- Acquiring MSYS Packages...
-- Acquiring MSYS Packages... OK
-- Fixing pkgconfig - release
-- Checking file: D:/dev/vcpkg/packages/zlib_x64-windows/lib/pkgconfig/zlib.pc
-- Fixing pkgconfig - debug
-- Checking file: D:/dev/vcpkg/packages/zlib_x64-windows/debug/lib/pkgconfig/zlib.pc
-- CHECK_LIB_zlibd_DEBUG:CHECK_LIB_zlibd_DEBUG-NOTFOUND
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:239 (message):
  Library "zlibd" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:348 (vcpkg_fixup_pkgconfig_check_files)
  ports/zlib/portfile.cmake:46 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:79 (include)


[DEBUG] cmd_execute_and_stream_data() returned 1 after  5167701 us
Error: Building package zlib:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zlib:x64-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.
[DEBUG] D:\DEV\vcpkg\toolsrc\src\vcpkg\install.cpp(494)
[DEBUG] Exiting after 7119387 us (7117175 us)
```

**Comment by Neumann-A** [comment]
D:\DEV\vcpkg\toolsrc\src\vcpkg\install.cpp(494)

versus 

D:/dev/vcpkg/packages/zlib_x64-windows/debug/lib/pkgconfig/zlib.pc

duplicated of #13105
closed by #13144

**Comment by NancyLi1013** [comment]
Hi @samsaga2 
Thanks for posting this issue.

Let's track this issue there #13105.

@Neumann-A 
Thanks for your information.
